lib/cylc/config.py

```diff
@@ -1380,12 +1391,15 @@ def generate_taskdefs( self, line, left_nodes, right, section, seq,
                     self.taskdefs[ name ].add_sequence(seq)
 
             if self.run_mode == 'live':
-                # register any explicit internal outputs
-                if 'outputs' in self.cfg['runtime'][name]:
-                    for lbl,msg in self.cfg['runtime'][name]['outputs'].items():
-                        outp = output(msg, base_interval)
-                        self.taskdefs[name].outputs.append(outp)
-
+                # register explicit output messages
+                for lbl,msg in self.cfg['runtime'][name]['outputs'].items():
```

Comment:
I guess the if isn't being used any more as iterating over nothing will have the same end result?